AFHTTPClientLogger.m

```diff
@@ -40,24 +57,42 @@ @implementation AFHTTPClientLogger
 
 - (instancetype)initWithBaseURL:(NSURL *)baseURL {
     if ((self = [super init])) {
-        self.baseURLString = [baseURL absoluteString];
-        self.level = AFHTTPClientLogLevelInfo;
+        _baseURLString = [baseURL absoluteString];
+        _level = AFHTTPClientLogLevelInfo;
+        _notificationHandlerQueue = [[NSOperationQueue alloc] init];
     }
 
     return self;
 }
 
 - (void)dealloc {
-    [[NSNotificationCenter defaultCenter] removeObserver:self];
+    [_notificationHandlerQueue cancelAllOperations];
+    [[NSNotificationCenter defaultCenter] removeObserver:_startNotificationObserver name:AFNetworkingOperationDidStartNotification object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:_finishNotificationObserver name:AFNetworkingOperationDidFinishNotification object:nil];
 }
 
 - (void)setEnabled:(BOOL)enabled {
     if (enabled != _enabled) {
         if (enabled) {
-            [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(operationDidStart:) name:AFNetworkingOperationDidStartNotification object:nil];
-            [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(operationDidFinish:) name:AFNetworkingOperationDidFinishNotification object:nil];
+            self.startNotificationObserver = [[NSNotificationCenter defaultCenter] addObserverForName:AFNetworkingOperationDidStartNotification
+                                                                object:nil
+                                                                 queue:self.notificationHandlerQueue
+                                                            usingBlock:^(NSNotification * _Nonnull notification) {
+                                                                [self operationDidStart:notification];
```

Comment:
Won't this retain `self` for the lifetime of the notification observer? We remove the observer in `-dealloc`, but this might prevent `-dealloc` from ever being called.